usl_pipeline/cloud_functions/main.py

```diff
@@ -149,30 +151,59 @@ def wrapper(cloud_event: functions_framework.CloudEvent) -> None:
 def build_and_upload_study_area_chunk(
     cloud_event: functions_framework.CloudEvent,
 ) -> None:
-    """Creates a study area chunk and uploads it to GCS.
+    """Creates a study area chunk, uploads to bucket, and writes to metastore.
 
     This function is triggered when files containing raw geo data for a study area
-    are uploaded to GCS. It will group the files into TAR-files and upload it to a
-    bucket.
+    are uploaded to GCS. It will load the file into study area chunks bucket and
+    write a new study area to metastore.
     """
-    file_name = cloud_event.data["name"]
+    file_name = pathlib.PurePosixPath(cloud_event.data["name"])
     bucket_name = cloud_event.data["bucket"]
 
-    if re.search(file_names.WPS_DOMAIN3_NC_REGEX, file_name):
+    # For AtmoML, only 3rd domain (500m) output files will be used
+    if re.search(file_names.WPS_DOMAIN3_NC_REGEX, file_name.name):
```

Comment:
Will we have multiple uploads of multiple uploads of different files for the same area, thus causing this to trigger multiple times and recreate the same study area?

That's fine, but might be worth documenting.

Is there an easy way to only run this function once? (Check if the study area is already in the metatore, or only trigger on file 0 or such?)

Reply:
Yeah that's right - I originally had a check in here to check if the study area existed before writing but you had mentioned that the multiple writes shouldn't be too big a deal. personally, i'd prefer if we'd only write once

Reply:
it really is not a big deal to set multiple times -- would only suggest changing things if it's a really easy change
I think it's fine to ride with what you've got down already